### ISSUE TITLE
BUG: IP Address Formatting in DB

### ISSUE THREAD

**Issue by nelsonic** [dwyl/hits#153]
On our catch up call, @SimonLab noted that the IP addresses are not being stored correctly.

```sh
flyctl postgres connect -a hits-db
```

```sh
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
 0.0.0.0.0.65535.47400.59506
--More--
```

Ref: https://en.wikipedia.org/wiki/0.0.0.0

![image](https://user-images.githubusercontent.com/194400/157851670-8512dbeb-cc8e-4009-adab-ceefd2cedda3.png)

I suspect this might be a Fly.io thing ... 

# Todo

+ [ ] Investigate why the IP addresses aren't being stored correctly by _briefly_ logging them to `sdout` (`IO.inspect`) so that we can simply tail the logs and see if the issue is at the database level.
  + [ ] Report back your findings!

**Comment by nelsonic** [comment]
The reason we want to fix this ASAP is so that we can start doing Geo IP lookups for #27 🗺️

**Comment by SimonLab** [comment]
from: https://elixirforum.com/t/how-to-show-clients-ipv4-on-page/41110

>By default, Plug puts the IP of the peer in conn.remote_ip. When running behind a load balancer, that will be the internal IP of the load balancer.

Could be due to https://fly.io/docs/about/pricing/#anycast-ip-addresses maybe

**Comment by nelsonic** [comment]
Good detective work! 🔍  👍 
So this is the IP address of the load balancer forwarding the requests ... 🤦‍♂️ 
Please do some `IO.inspect` logging to find out if we can get the client IP. 🤞

**Comment by nelsonic** [comment]
Yeah, that's probably enough. I was just thinking that they might have a block of IP addresses that `github-camo` makes outbound requests from and we could just ignore them completely if the `ua` is `github-camo`. But what we _already_ have is fine. 👍 
Let's not spend any more time on this from the GitHub Hits perspective and simply move onto other higher priority tasks.
It would be good to store the full useragent + IP data for other pages e.g: https://tachyons-bootstrap.dwyl.com/ or at least check if that page, given that it's hosted by GitHub pages, is proxying the hits badge requests ... 💭

**Comment by nelsonic** [comment]
@SimonLab nice one. Thanks. 👍
